model/githubapp/github_app_installation.go

```diff
@@ -323,17 +266,22 @@ func createCacheID(installationID int64, permissions *github.InstallationPermiss
 	// be concatenated into the cache ID.
 	for i := 0; i < permissionsStructVal.NumField(); i++ {
 		field := permissionsStructVal.Field(i)
-		if field.Kind() != reflect.Ptr || field.Elem().Kind() != reflect.String {
+		if field.Kind() != reflect.Ptr ||
+			field.Elem().Kind() != reflect.String ||
+			field.IsNil() {
 			continue
 		}
 
-		if !field.IsNil() {
-			fieldValue, ok := field.Interface().(*string)
-			if !ok {
-				return "", errors.New("failed to convert permission field to string")
-			}
-			permissionPairs = append(permissionPairs, fmt.Sprintf("%s:%s", permissionsStructVal.Type().Field(i).Name, utility.FromStringPtr(fieldValue)))
+		fieldValue, ok := field.Interface().(*string)
+		if !ok {
+			return "", errors.Errorf(
+				"expected *string for field '%s', got '%T' with value '%v'",
+				permissionsStructVal.Type().Field(i).Name,
+				reflect.TypeOf(field.Interface()),
+				field.Interface(),
```

Comment:
I added some extra information in case we ever need to debug it, but lmk if it's too much information